app/frontend/src/components/ches/HistoryTable.vue

```diff
@@ -57,68 +59,113 @@
           </template>
           <span>Cancel Scheduled Send</span>
         </v-tooltip>
-      </template> -->
+      </template>
+
+      <template v-slot:[`footer.prepend`]>
+        <v-btn color="primary" small class="ma-2" @click="populateTable">
+          Refresh<v-icon small class="ml-2">refresh</v-icon>
+        </v-btn>
+        <v-btn color="primary" small class="ma-2" @click="clearTable">
+          Clear<v-icon small class="ml-2">clear</v-icon>
+        </v-btn>
+      </template>
     </v-data-table>
   </v-container>
 </template>
 
 <script>
 import { format } from 'date-fns';
 import { mapActions, mapGetters } from 'vuex';
+import { mapFields } from 'vuex-map-fields';
 
+import chesService from '@/services/chesService';
 export default {
   name: 'HistoryTable',
   data: () => ({
     error: false,
-    loading: false,
     search: '',
     headers: [
       { text: 'Transaction ID', align: 'start', value: 'txId' },
       { text: 'Message ID', align: 'start', value: 'msgId' },
       { text: 'Tag', align: 'start', value: 'tag' },
       { text: 'Status', align: 'start', value: 'status' },
       { text: 'Delayed', align: 'start', value: 'delayTS', width: 150 },
-      // TODO: implement promote and cancel actions
-      // {
-      //   text: 'Actions',
-      //   align: 'end',
-      //   value: 'actions',
-      //   filterable: false,
-      //   sortable: false,
-      // },
+      { text: 'Actions', align: 'start', value: 'actions', width: 120, filterable: false, sortable: false }
     ],
   }),
 
   computed: {
-    ...mapGetters('ches', ['txIds', 'tableData']),
+    ...mapGetters('ches', ['txIds']),
+    ...mapFields('ches', ['tableData', 'tableLoading'])
   },
 
   methods: {
     ...mapActions('alert', ['showAlert', 'clearAlert']),
-    ...mapActions('ches', ['addTableData', 'populateTable']),
+    ...mapActions('ches', ['populateTable', 'clearTable']),
+
 
-    cancelMessage(msgId) {
-      console.log('cancelMessage', msgId); // eslint-disable-line no-console
+    async cancelMessage(msgId) {
+      try {
+        // wait for OK status from CHES api
+        await chesService.cancel(msgId);
+        // update store with latest table data
+        this.populateTable();
+        // show success alert
+        this.showAlert({
+          type: 'success',
+          text: `<strong>Your message has been successfully cancelled.<br />Message ID: </strong> ${msgId}`,
+        });
+      } catch (e) {
+        this.error = true;
+        // show error alert
+        this.showAlert({
+          type: 'error',
+          text: e,
+        });
+      }
+      window.scrollTo(0, 0);
```

Comment:
Should we be forcing the page to the top after this operation? I suspect this is meant for situations where the user decides to have a large number of rows on the page and we want to make sure that if a notification shows up they can see it. Should be fine but will defer to other team members on what they think about this.

Reply:
Is the success alert at the top of the page? Or always on the screen? Would be good to make sure message is shown so if it's pinned then scroll seems appropriate.
If floating then yeah could leave page where it is